woocommerce/woocommerce
# Fix Stock Notice behavior in the Single Product block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Partially addresses https://github.com/woocommerce/woocommerce/issues/53011.

![Screenshot 2024-11-29 at 5 12 11 PM](https://github.com/user-attachments/assets/e3d3fe63-428b-471a-b2a9-4a5680e867e6)

This PR adds code to determine the availability of the stock indicator based on the following criteria:

- **Product Type:**
  - The stock indicator is only displayed for allowed products: simple products and single variations with `stock management` enabled.
  - The indicator will not be shown when `stock management` is disabled and the product is in stock.
  
These changes ensure consistent visibility of the stock indicator in the `Single Product` block in the editor and on the front end.
This PR also adds the component `Stock Indicator` to the inner blocks within the `Add to Cart with Options (experimental)` block.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Having `WooCommerce Beta Tester` installed and activater
1. Go to Tools > WCA Test Helper > Features
2. Enable the features: `blockified-add-to-cart` and `experimental-blocks`
3. Create some different products (Simple, Variable, Grouped, External).
4. Go to Pages > Add New Page
5. Add a `Single Product` block and select a simple product.
6. Add a `Stock Indicator` block inside the `Single Product` block.
7. The stock indicator won't be visible for single products when:
- The `manage stock` is disabled and is in stock
- The `manage stock` is enabled, quantity is 0 and `Allow backorders?` is Allow
8. Now select a variable product.
9. Pick one of the variations from the variable product you chose before.
- The stock indicator won't be visible when:
  - The parent has `manage stock`, quantity is 0 and `Allow backorders?` is `Allow`. And the picked variation doesn't manage stock.
  - The parent don't has `manage stock` and in the variation is `in stock` is selected.
10. The stock indicator won't be visible for `grouped` and `external` products.
11. Verify that no regressions have been introduced.

## UPDATE

For reference, the product page and the page you just created with the `Single Product` block should display the same stock indicator behavior.




### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
